tensorboard/plugins/mesh/metadata.py

```diff
@@ -18,15 +18,35 @@
 from __future__ import division
 from __future__ import print_function
 
+from collections import namedtuple
 from tensorboard.compat.proto import summary_pb2
 from tensorboard.plugins.mesh import plugin_data_pb2
 
 
+MeshTensor = namedtuple('MeshTensor', 'data content_type data_type')
 PLUGIN_NAME = 'mesh'
 
 # The most recent value for the `version` field of the
 # `MeshPluginData` proto.
-_PROTO_VERSION = 1
+_PROTO_VERSION = 0
+
+
+def get_components_bitmask(tensors):
+  """Creates bitmask for all existing components of the summary.
+
+  Args:
+    tensors: list of MeshTensor tuples, representing all components
```

Comment:
Taking the whole `MeshTensor` when all you need is the `content_type`
seems unnecessary. The `tensor.data is None` effectively turns this
function from “reduction operation” to “reduction operation plus
filter”, which feels superfluous. If you change this to accept the
content types only, then the client in `summary.py` can simply write

```python
components = metadata.get_components_bitmask(
    tensor.content_type for tensor in tensors
    if tensor.data is not None
)
```

and the client in `parse_plugin_metadata` no longer has to construct
“fake” mesh tensors with empty dicts as `data` just to use this function
because the responsibilities have been more cleanly separated.

Reply:
good idea